File: src/main/java/net/gtaun/shoebill/event/actor/ActorEvent.java
package net.gtaun.shoebill.event.actor;

import net.gtaun.shoebill.object.Actor;
import net.gtaun.util.event.Event;

/**
 * Base class for all ActorEvents, such as {@link net.gtaun.shoebill.event.actor.ActorStreamInEvent}
 *
 * @author Marvin Haschker
 * @see net.gtaun.util.event.Event
 */
public abstract class ActorEvent extends Event {

    private Actor actor;

    public ActorEvent(Actor actor) {
        this.actor = actor;
    }

    public Actor getActor() {
        return actor;
    }
}

File: src/main/java/net/gtaun/shoebill/event/actor/ActorEvent.kt
package net.gtaun.shoebill.event.actor

import net.gtaun.shoebill.entities.Actor
import net.gtaun.util.event.Event

/**
 * Base class for all ActorEvents.
 *
 * @author Marvin Haschker
 */
open class ActorEvent(val actor: Actor) : Event() {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ActorEvent) return false

        if (actor != other.actor) return false

        return true
    }

    override fun hashCode(): Int {
        return actor.hashCode()
    }

}
